### PR TITLE
boards: arm: stm32f3 disco has stm32cubeprogrammer as runner

### PR DESCRIPTION
**Pull request by FRASTM**
Adds the stm32cubeprogrammer as runner of the stm32f3_disco
This will help testing the platform and others with twister and more

Change the ../map.yaml if needed

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: boards/arm/stm32f3_disco/board.cmake [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 board_runner_args(jlink "--device=STM32F303VC" "--speed=4000")
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset=hw")
 
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
```
Comment (gmarull) [inline boards/arm/stm32f3_disco/board.cmake:6]: why is this added as the first runner? stm32cubeprogrammer does not allow e.g. to debug

**Comment by gmarull** [comment]
> Adds the stm32cubeprogrammer as runner of the stm32f3_disco This will help testing the platform and others with twister and more

why can't twister be used now?

File: boards/arm/stm32f3_disco/board.cmake (hunk 1 above)
Comment (FRASTM) [reply boards/arm/stm32f3_disco/board.cmake:6]: No particular reason, it was just for convenience on my own environment.
--> putting to the last entry

**Comment by FRASTM** [comment]
> > Adds the stm32cubeprogrammer as runner of the stm32f3_disco This will help testing the platform and others with twister and more
> 
> why can't twister be used now?

We had some situations were the stm32f3_disco reported "Device issue (Flash error?)" when running.

**Review by gmarull — APPROVED** [review]

**Review by erwango — APPROVED** [review]
